### ISSUE TITLE
[Bug]: unnecessary steps, forget existing files, overwrites files

### ISSUE THREAD

**Issue by fblgit** [Pythagora-io/gpt-pilot#487]
### Version

VisualStudio Code extension

### Operating System

Ubuntu Linux

### What happened?

Overall:
- It overwrite files, loosing data on iterations
- Human made changes gets lost on iterations
- Polluting the folder with existing methods, etc
- Dozens of unnecessary steps, and they cannot be jumped
- Output text, then output JSON.. double generation/half-cut context as result

My suggestions are:
- Before modifying a file, retrieve it and ask to adapt the existing file
- Prompting: exclusive JSON output, avoid text to JSON duplication
- Keep a permanent state of the filetree, methods, functions ,inputs, outputs within the context as a permanent element.
- Chain GPT3.5 to provide basics like "shall we refector?, shall we XXXX ?" so it can be used to cheap & quick data binary classification kind of.

Since the code itself doesn't have a logic per-se and it leverages on agents to operate, it should have different tiers where use-case/purpose can be scoped (ex: QuestionAnswering: gpt3.5-turbo, JSONtoNL: gpt3.5-turb,  troubleshoot: gpt4-turbo, etc). If we can integrate this with API GPTs/Agents/Langchain.. so the interpreter/sandbox/etc can be fully leveraged... it may take him 1 hour to make a CRUD.. but, he may actually do it 🍡 

Overall, its a bruteforcer.. but at least it can deliver "something"..  its also important to have better integration with other free-models.. its not like u wanna spend money on create a python env or pip install a package..

**Comment by SeKi81** [comment]
Yes, he does that in pretty much every project. It makes every project unusable.
